crates/core/tedge/src/error.rs

```diff
@@ -27,4 +27,10 @@ pub enum TEdgeError {
 
     #[error(transparent)]
     FromTEdgeConfigRead(#[from] tedge_config::new::ReadError),
+
+    #[error(transparent)]
+    FromConfigNotSet(#[from] tedge_config::new::ConfigNotSet),
+
+    #[error(transparent)]
+    CertificatesNotSet(#[from] tedge_config::CertificateError),
```

Comment:
I don't see this error variant used anywhere, so it can probably be deleted.

Reply:
This error is used here     `config.mqtt_client_auth_configs()?`

Reply:
Ah right, you're correct in that this variant would be constructed using the `From` conversion, had `mqtt_client_auth_configs()` returned an `Err(CertificateError)`, but looking at the function body, this function does not return an error in any branch, so as it is now, the `Result` there is useless, making this variant here useless as well.

Reply:
Yes, removed.